src/main/java/com/internetitem/logback/elasticsearch/ClassicElasticsearchPublisher.java

```diff
@@ -40,10 +36,29 @@ protected void serializeCommonFields(JsonGenerator gen, ILoggingEvent event) thr
             gen.writeObjectField("message", formattedMessage);
         }
 
-        if(settings.isIncludeMdc()) {
+        if (settings.isIncludeMdc()) {
+            List<String> excludedKeys = getExcludedMdcKeys();
             for (Map.Entry<String, String> entry : event.getMDCPropertyMap().entrySet()) {
-                gen.writeObjectField(entry.getKey(), entry.getValue());
+                if (!excludedKeys.contains(entry.getKey())) {
+                    gen.writeObjectField(entry.getKey(), entry.getValue());
+                }
+            }
+        }
+    }
+
+    private List<String> getExcludedMdcKeys() {
+        /*
+         * using a List instead of a Map because the assumption is that
+         * the number of excluded keys will be very small and not cause
+         * a performance issue
+         */
+        List<String> result = new ArrayList<>();
+        if (settings.getExcludedMdcKeys() != null) {
+            String[] parts = settings.getExcludedMdcKeys().split(",");
+            for (String part : parts) {
+                result.add(part.trim());
             }
         }
+        return result;
```

Comment:
Is there a reason this list is re-allocated and populated anew every time a log entry is written? Why is this not parsed once when reading the config and then re-used on subsequent accesses? Looks like this could incur a performance overhead.

List instead of a map or set is probably fine, because this will likely only matter when there are hundreds of items in the collection and calculating the hash code of each tested key can be avoided.